### ISSUE TITLE
Alert component: Help teams handle accessibility roles in the right situation

### ISSUE THREAD

**Issue by davidakennedy** [department-of-veterans-affairs/vets-design-system-documentation#1189]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

I had a team that added the `role="alert"` attribute to a heading in the Alert component. The team had the right idea here since they needed the Veteran to know about this error right away. However, placing it on a heading component cancels the heading semantics of that element. 

## What I expected to happen

How might engineers better know about the different roles potentially needed in the Alert component, and which one is best? I would have expected the team to add that role to the outer container of the component. 

## Urgency
How urgent is this request? Please select the appropriate option below and/or provide details
- [ ] This bug is blocking work currently in progress
- [ ] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [x] This bug is not blocking any work
- [ ] Other

### Details

We had a [brief discussion on this on Slack](https://dsva.slack.com/archives/C01DBGX4P45/p1662649470552819). Some ideas for possible directions here:

- We could add documentation to the site, explaining which ARIA roles make the most sense for certain situations and rely on teams to add them.
- We could add a property to the Alert component that would enable teams to add different roles to the outer container of the component. Like: `'alert' | 'alertdialog' | 'region' | 'status'`
- We could do a bit of both of the above, adding the ability to add the role to the component, and some documentation around which role is best for different situations. It's highly contextual, depending on the content, the page, etc. This third options may be the sweet spot.

I also want to note that we may be able to further improve the screen reader experience when these roles are added by using `aria-labelledby` with the headings. It's something we could dig into later.

**Comment by davidakennedy** [comment]
@SarahKay8 This is a great start! I do think adding some short documentation is a good place to begin here. I read back over the Slack thread, and I would be hesitant to add a role to this component by default. Only because there could be a lot of different situations where an alert could be used. I took what you drafted, and expanded it. Feedback welcome!

"In some situations, an ARIA role may need to be added to the alert component for it to work best for people who use assistive technology. ARIA should be used sparingly to supplement and enhance the native features of HTML.

If the alert is a static alert that exists on the page when the page gets loaded, it likely doesn't need a role.

If the alert conveys timely information that needs to be acted on before moving forward, use `role="alert"`. If the information in the alert contains interactive elements, like links or buttons, use the `role="alertdialog"` role instead.

If the information isn't timely and doesn't need to be acted on immediately, but still contains important information that represents a substantial topic on the page, `role="region"` may be appropriate.

Some roles, like `role="region"`, help assistive technology identify the content as being grouped and separated from the rest of the page content. For example, some assistive technologies can navigate by landmark roles like `role="region"`."

I think we could include links to maybe the [MDN docs about the specific roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques). What do you think?

Also curious if we need the part about `role="region"`? Do we have any use cases for that?

**Comment by SarahKay8** [comment]
Documentation updated! Thanks @davidakennedy. I included the <role-region> section (I believe it doesn't hurt to add) and added MDN links.

**Comment by humancompanion-usds** [comment]
Resolution here was to add this content:
https://design.va.gov/components/alert#assign-an-appropriate-aria-role
